lib/g10_certification_test_kit/bulk_data_group_export.rb

```diff
@@ -0,0 +1,258 @@
+    end
+
+    http_client :bulk_server do
+      url :bulk_server_url
+    end
+
+    # TODO: Create after implementing TLS Tester Class.
+    test do
+      title 'Bulk Data Server is secured by transport layer security'
+      description <<~DESCRIPTION
+        [§170.315(g)(10) Test Procedure](https://www.healthit.gov/test-method/standardized-api-patient-and-population-services) requires that
+        all exchanges described herein between a client and a server SHALL be secured using Transport Layer Security (TLS) Protocol Version 1.2 (RFC5246).
+      DESCRIPTION
+      # link 'http://hl7.org/fhir/uv/bulkdata/export/index.html#security-considerations'
+
+      run { pass }
+    end
+
+    test do
+      title 'Bulk Data Server declares support for Group export operation in CapabilityStatement'
+      description <<~DESCRIPTION
+        The Bulk Data Server SHALL declare support for Group/[id]/$export operation in its server CapabilityStatement
+      DESCRIPTION
+      # link 'http://hl7.org/fhir/uv/bulkdata/OperationDefinition-group-export.html'
+
+      run do
+        fhir_get_capability_statement(client: :bulk_server)
+        assert_response_status([200, 201])
+
+        assert_valid_json(request.response_body)
+        capability_statement = FHIR.from_contents(request.response_body)
+
+        definition = 'http://hl7.org/fhir/uv/bulkdata/OperationDefinition/group-export'
+
+        capability_statement.rest&.each do |rest|
+          groups = rest.resource&.select { |resource| resource.type == 'Group' }
+
+          pass if groups&.any? do |group|
+            group.operation&.any? do |operation|
+              if operation.definition.is_a? String
+                operation.definition == definition
+              else
+                operation.definition&.flatten&.include?(definition)
+              end
+            end
+          end
+        end
+
+        assert false, 'Server CapabilityStatement did not declare support for export operation in Group resource'
+      end
+    end
+
+    test do
+      title 'Bulk Data Server rejects $export request without authorization'
+      description <<~DESCRIPTION
+        The FHIR server SHALL limit the data returned to only those FHIR resources for which the client is authorized.
+
+        [FHIR R4 Security](http://build.fhir.org/security.html#AccessDenied) and
+        [The OAuth 2.0 Authorization Framework: Bearer Token Usage](https://tools.ietf.org/html/rfc6750#section-3.1)
+        recommend using HTTP status code 401 for invalid token but also allow the actual result be controlled by policy and context.
+      DESCRIPTION
+      # link 'http://hl7.org/fhir/uv/bulkdata/export/index.html#bulk-data-kick-off-request'
+
+      include ExportKickOffPerformer
+
+      run do
+        skip_if bearer_token.blank?, 'Could not verify this functionality when bearer token is not set'
+
+        perform_export_kick_off_request(use_token: false)
+        assert_response_status(401)
```

Comment:
The descriptions says `recommend using HTTP status code 401` but the test only checks for `401`

It looks like Inferno-Program allowed 400 or 401

https://github.com/onc-healthit/inferno-program/blob/a773c8ed407875b9614e11f692e51f350a6c2f6b/lib/modules/onc_program/bulk_data_export_sequence.rb#L188

https://github.com/onc-healthit/inferno-program/blob/a773c8ed407875b9614e11f692e51f350a6c2f6b/lib/app/utils/assertions.rb#L41

Reply:
Good catch.